pubsub/gcp/pubsub/pubsub.go

```diff
@@ -175,11 +176,12 @@ func (g *GCPPubSub) Publish(ctx context.Context, req *pubsub.PublishRequest) err
 		return errors.New("component is closed")
 	}
 
-	if !g.metadata.DisableEntityManagement {
+	if !g.metadata.DisableEntityManagement && !g.topicCache[req.Topic] {
 		err := g.ensureTopic(ctx, req.Topic)
 		if err != nil {
 			return fmt.Errorf("%s could not get valid topic %s, %s", errorMessagePrefix, req.Topic, err)
 		}
+		g.topicCache[req.Topic] = true
```

Comment:
All of these calls that read or set in the cache are not concurrent-safe and have race conditions.

Please use a sync.RWMutex around them.

Reply:
Good catch